src/storage/object/delete.js

```diff
@@ -9,35 +9,54 @@
  * OF ANY KIND, either express or implied. See the License for the specific language
  * governing permissions and limitations under the License.
  */
+import { deleteFromCollab } from '../utils/collab.js';
+import { postObjectVersionWithLabel } from '../version/put.js';
 
+/**
+ * Deletes an object in the storage, creating a version of it if necessary.
+ *
+ * @param {Object} env the CloudFlare environment
+ * @param {DaCtx} daCtx the DA Context
+ * @param {String} key the key of the object to delete (excluding Org)
+ * @param {Boolean} isMove if this was initiated by a move operation
+ * @return {Promise<void>}
+ */
+export async function deleteObject(env, daCtx, key, isMove = false) {
+  const fname = key.split('/').pop();
+
+  const tmpCtx = { ...daCtx, key }; // For next calls, ctx needs the passed key, as it could contain a folder
+  if (fname.includes('.') && !key.endsWith('.props')) {
```

Comment:
We should also check that fname does not _start with_ a dot, as there are hidden directories that do.